core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMaterializedViewDefinition.java

```diff
@@ -27,16 +27,30 @@
 {
     private final String originalSql;
     private final Optional<CatalogSchemaTableName> storageTable;
+    private final Optional<Boolean> delegateRefreshToConnector;
     private final Optional<String> catalog;
     private final Optional<String> schema;
     private final List<Column> columns;
     private final Optional<String> comment;
     private final String owner;
     private final Map<String, Object> properties;
 
+    public ConnectorMaterializedViewDefinition(
+            String originalSql,
+            Optional<String> catalog,
+            Optional<String> schema,
+            List<Column> columns,
+            Optional<String> comment,
+            String owner,
+            Map<String, Object> properties)
+    {
+        this(originalSql, Optional.empty(), Optional.empty(), catalog, schema, columns, comment, owner, properties);
+    }
+
     public ConnectorMaterializedViewDefinition(
             String originalSql,
             Optional<CatalogSchemaTableName> storageTable,
+            Optional<Boolean> delegateRefreshToConnector,
```

Comment:
Why is this Optional? What's the meaning when it's `Optional.empty()`?

Reply:
`ConnectorMaterializedViewDefinition` is used by engine when creating MV, but it's also used by connector when returning MV.

Essentially, `delegateRefreshToConnector` is an output field similar as `storageTable`. When engine creates MV, these two fields are `empty` since engine doesn't know connector preference and storage table.

Reply:
That is a little confusing and it indicates to me that there's something odd going on with the API. It may be more appropriate to add a separate API call to obtain that kind of information about the view.

Reply:
I think changing SPI is out of scope of this PR. It would have to address both `delegateRefreshToConnector` and `storageTable`. We would most likely have to introduce some `ConnectorMaterializedViewHandle` to avoid repeated metadata DB calls withing connector (alternative would be matadata caching in connectors).

I think keeping this "output" fields within `ConnectorMaterializedViewDefinition` could do for now unless we see some blockers there

Reply:
added `isDelegateMaterializedViewRefreshToConnector`